homeassistant/components/telegram_bot/__init__.py

```diff
@@ -137,7 +139,18 @@
                         ),
                         vol.Required(CONF_API_KEY): cv.string,
                         vol.Required(CONF_ALLOWED_CHAT_IDS): vol.All(
-                            cv.ensure_list, [vol.Coerce(int)]
+                            cv.ensure_list,
+                            [
+                                vol.Any(
+                                    vol.Coerce(int),
+                                    vol.Schema(
+                                        {
+                                            vol.Required(CONF_CHAT_ID): vol.Coerce(int),
+                                            vol.Optional(CONF_USER_ID): cv.string,
```

Comment:
This format seems a bit awkward, would it be better to either allow a list of chat ids, or a map from chat id to user ids?
Also, which user id is this, how do users find it?

Reply:
I will try to change the format in a separate PR.
The user id is the home assistant user id, the user can find it in Developer tools in the person entity state attributes.